### PR TITLE
Feature: KeycloakDeploymentSpec.PodSecurityContext

### LINKED ISSUE #458: Error: container has runAsNonRoot and image will run as root [securityContext]
### Description

Is there a possibility to apply a `securityContext` to the Keycloak deployment via the Operator?

My deployment failes with the Error message: `Error: container has runAsNonRoot and image will run as root`

### Discussion

_No response_

### Motivation

Security. Containers should not run as root.

### Details

_No response_

### PR DESCRIPTION
**Pull request by Nold360**
## Related GH Issue
Closes #458

## Additional Information
Adds PodSecurityContext to Keycloak with default values:
```yaml
runAsNonRoot: true
runAsUser: 1000
runAsGroup: 1000
fsGroup: 1000
```

## Checklist:
- [x] Automated Tests
- [ ] [Documentation](https://github.com/keycloak/keycloak-documentation) - follows if this PR has a chance to get merged

### CODE REVIEW

**Comment by andreaTP** [comment]
Hi @Nold360 , can you add some tests please?
I think that this PR: https://github.com/keycloak/keycloak-operator/pull/414/files is small enough and shows how to add tests in the relevant sections.

**Comment by Nold360** [comment]
I think i figured out this unit test thing :D

**Comment by andreaTP** [comment]
LGTM, cc. @vmuzikar

**Review by andreaTP — APPROVED** [review]

**Comment by Nold360** [comment]
Fixed the commit message

**Comment by vmuzikar** [comment]
Closing and re-opening to fix the CI.

File: pkg/model/keycloak_deployment.go [hunk 1]
```diff
@@ -28,6 +28,22 @@ func GetServiceEnvVar(suffix string) string {
...
+
+	defaultUserAndGroup := int64(1000)
+	runAsNonRoot := true
+
+	return &v1.PodSecurityContext{
+		RunAsUser:    &defaultUserAndGroup,
+		RunAsGroup:   &defaultUserAndGroup,
+		RunAsNonRoot: &runAsNonRoot,
+		FSGroup:      &defaultUserAndGroup,
+	}
```
Comment (vmuzikar) [inline pkg/model/keycloak_deployment.go:44]: This is potentionally a breaking change, we weren't setting any defaults before. Could you please remove it?

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@Nold360 Thank you for your contribution! Added a few comments.

**Comment by vmuzikar** [comment]
@Nold360 Just wanted to ask if this is still something you'd like to work on.

**Comment by vmuzikar** [comment]
@Nold360 Closing this for now. Feel free to reopen!

**Pull request closed without merging**